### PR TITLE
Add net.sonic_pi.SonicPi

### PR DESCRIPTION
**Pull request by hfiguiere**
This add SonicPi.

There is probably some polish needed.

Requires pipewire fiddling too.

Upstream issue: https://github.com/sonic-pi-net/sonic-pi/issues/2022

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build net.sonic_pi.SonicPi

**Comment by hfiguiere** [comment]
bot, build net.sonic_pi.SonicPi

**Comment by hfiguiere** [comment]
bot, build net.sonic_pi.SonicPi

**Comment by hfiguiere** [comment]
bot, build net.sonic_pi.SonicPi

**Comment by hfiguiere** [comment]
(the previous build was broken, ie it didn't work)

**Comment by hfiguiere** [comment]
bot, build net.sonic_pi.SonicPi

File: net.sonic_pi.SonicPi.json [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    "runtime": "org.kde.Platform",
+    "sdk": "org.kde.Sdk",
+    "runtime-version": "5.15",
+    "command": "sonic-pi",
+    "rename-icon": "sonic-pi",
+    "rename-desktop-file": "sonic-pi.desktop",
+    "finish-args": [
+        /* X11 + XShm access */
+        "--share=ipc",
+        "--socket=x11",
```
Comment (barthalion) [inline net.sonic_pi.SonicPi.json:14]: Should be `fallback-x11` if it works natively on Wayland.

**Comment by barthalion** [comment]
bot, build net.sonic_pi.SonicPi

**Comment by barthalion** [comment]
@hfiguiere Is there anything left before we can merge it?

File: net.sonic_pi.SonicPi.json (hunk 1 above)
Comment (hfiguiere) [reply net.sonic_pi.SonicPi.json:14]: I actually now disabled Wayland, it works better, like a lot of Qt apps.

**Comment by hfiguiere** [comment]
bot, build net.sonic_pi.SonicPi

**Comment by hfiguiere** [comment]
> @hfiguiere Is there anything left before we can merge it?

Most of the issues are WTR to JACK support and pipewire. It works, but definitely still experimental.

**Comment by hfiguiere** [comment]
> > @hfiguiere Is there anything left before we can merge it?
> 
> Most of the issues are WTR to JACK support and pipewire. It works, but definitely still experimental.

Just to clarify, issues are outside flatpak.

**Comment by barthalion** [comment]
We can merge it to beta but I leave the decision to you.

**Comment by hfiguiere** [comment]
Let's merge on stable. When pipewire is configured, it works.

**Comment by barthalion** [comment]
Hi @samaaron, if you don't mind, we will send you an invitation if you ever want to change something in the Flathub package yourself.

/merge @samaaron

**Comment by samaaron** [comment]
Ace, thanks for all your hard work!

**Comment by barthalion** [comment]
It didn't work for some reason, let's try again.

/merge @samaaron

**Comment by barthalion** [comment]
The universe says no:

```
Cloning https://github.com/hfiguiere/flathub.git (branch: sonicpi)
  File "/entrypoint.py", line 182, in <module>
    main()
  File "/entrypoint.py", line 126, in main
    clone.update_submodules(init=True)
  File "/usr/lib/python3.9/site-packages/pygit2/repository.py", line 162, in update_submodules
    payload.check_error(err)
  File "/usr/lib/python3.9/site-packages/pygit2/callbacks.py", line 93, in check_error
    check_error(error_code)
  File "/usr/lib/python3.9/site-packages/pygit2/errors.py", line 56, in check_error
    raise KeyError(message)
KeyError: 'object not found - no match for id (16cf6148decb4dd0b663ff8477b839646743f98d)'
```

I will perform the merge ritual manually tomorrow morning.

**Comment by barthalion** [comment]
Alright, done: https://github.com/flathub/net.sonic_pi.SonicPi

You can accept your invitations at https://github.com/flathub/net.sonic_pi.SonicPi/invitations. Thanks!

**Pull request closed without merging**